Userland/Libraries/LibWeb/Page/EventHandler.cpp

```diff
@@ -255,8 +255,11 @@ bool EventHandler::handle_mouseup(CSSPixelPoint position, unsigned button, unsig
             handled_event = true;
 
             bool run_activation_behavior = true;
-            if (node.ptr() == m_mousedown_target && button == GUI::MouseButton::Primary) {
-                run_activation_behavior = node->dispatch_event(UIEvents::MouseEvent::create_from_platform_event(node->realm(), UIEvents::EventNames::click, offset, client_offset, page_offset, button).release_value_but_fixme_should_propagate_errors());
+            if (node.ptr() == m_mousedown_target) {
+                if (button == GUI::MouseButton::Primary)
+                    run_activation_behavior = node->dispatch_event(UIEvents::MouseEvent::create_from_platform_event(node->realm(), UIEvents::EventNames::click, offset, client_offset, page_offset, button).release_value_but_fixme_should_propagate_errors());
+                else if (button == GUI::MouseButton::Secondary && !(modifiers & Mod_Shift)) // Allow the user to bypass custom context menus by holding shift, like Firefox.
```

Comment:
i feel that this modifier should be “shift only” (not sure whether this is the behaviour observable in wild, though, e.g. in firefox).